packages/constants/src/index.ts

```diff
@@ -1,3 +1,47 @@
+import { bn } from '@fuel-ts/math';
+
 export const ZeroBytes32 = '0x0000000000000000000000000000000000000000000000000000000000000000';
 export const NativeAssetId = ZeroBytes32;
 export const EmptyRoot = '0xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855';
+
+export const getEnv = () => ({
+  /** Maximum contract size, in bytes. */
+  CONTRACT_MAX_SIZE: 16 * 1024,
+
+  /** Maximum number of witnesses. */
+  MAX_WITNESSES: 16,
+
+  /** Maximum gas per transaction. */
+  MAX_GAS_PER_TX: bn(100000000),
+
+  /**
+   * Gas Price factor this is used to calculate
+   * This is used to calculate the gas fee in Native Coins.
+   * Ex.: transactionFee = Math.ceil(<gasUsed> / MAX_GAS_PER_TX);
+   */
+  GAS_PRICE_FACTOR: bn(1000000),
+
+  /** Gas charged per byte of the transaction. */
+  GAS_PER_BYTE: bn(4),
+
+  // TODO: set max script length const
+  /** Maximum length of script, in instructions. */
+  MAX_SCRIPT_LENGTH: 1024 * 1024 * 1024,
+
+  // TODO: set max script length const
+  /** Maximum length of script data, in bytes. */
+  MAX_SCRIPT_DATA_LENGTH: 1024 * 1024 * 1024,
+
+  /** Maximum number of static contracts. */
+  MAX_STATIC_CONTRACTS: 255,
+
+  // TODO: set max predicate length value
+  /** Maximum length of predicate, in instructions. */
+  MAX_PREDICATE_LENGTH: 1024 * 1024,
+
+  // TODO: set max predicate data length value
+  /** Maximum length of predicate data, in bytes. */
+  MAX_PREDICATE_DATA_LENGTH: 1024 * 1024,
+
+  FAILED_TRANSFER_TO_ADDRESS_SIGNAL: '0xffffffffffff0001',
+});
```

Comment:
Shouldn't the existent properties be moved inside the `getEnv` method?
 - `ZeroBytes32`
 - `NativeAssetId`
 - `EmptyRoot`

Reply:
Yes, from the outsider's perspective, it does feel weird.

Haven't looked at it from this angle.

 1. What would you suggest?
 1. Parallel to this, do you think the `constants` package is essential? Is it just a matter of moving things where they belong, or does the package solve a bigger issue? (e.g., cyclical refs)

Reply:
1. I suggest omitting the completely static variables from the `getEnv()` function just for the `constants` package to optimize for our SDK's users. These constants can live at the base level of the file as constants, and they can be hence easily imported. Let's use the `getEnv()` function only for values that actually rely on the environment they are being used in.

2. Yes, if there is no `constants` package there were several cases where I was facing cyclical refs/dependencies. That's really the only major reason why I think we need this package, apart from the fact that some of these constants are being used _outside_ of the package where they 'belong' that it somewhat makes sense to have them placed in a global store of some sort.

Reply:
I still think we can remove the duality _(of choosing where to put things)_ by **_eliminating_** the `constants` package.

We could solve the cyclic imports with isolated entry points:

```ts
// index.js
// the line below should not exist
// export * from "./configs";

// configs.js
export const ABC = 123

// package.json
"exports": {
  "./configs": {
    "require": "./dist/configs.js",
    "import": "./dist/configs.mjs",
    "types": "./dist/configs.d.ts"
  }
}
```

Other files can use it like:

```ts
import { ABC } from '@fuel-ts/<some-pkg>/configs';

console.log({ ABC })
```

The main `fuels` package can export everything like:

```ts
export * from '@fuel-ts/<some-pkg>/configs'
export * from '@fuel-ts/<other-pkg>/configs'
// ...
```

In short:
 1. Using the `configs` terminology _(instead of `consts` or `env`)_
 2. Exporting individual entry points for config files _(in `package.json`)_
 3. Not exporting configs in the packages' index file _(only on `fuels` index)_
 4. Removing `getEnv` methods and having the variables statically defined

Although we thought about encapsulating all variables inside methods such as `getEnv` in preparation for the future involving async variable fetching, I'm reconsidering it due to the same reason I cited at the beginning:
 - Removing duality

To make things easier, especially because **_most_** variables will **_not_** be fetched remotely, we could pull it from the inside out and do the opposite: have _all variables statically defined_ and use methods **_only_** for ones requiring async fetching.

This would leave a minimum footprint:

```ts
// configs.js
export const ABC = 123
export const XYZ = async () => fetch(...) // async stuff
```

All of this combined should provide the following:
 1. decoupled sync/async support
 1. freedom from cyclic-dependency errors
 1. coherence where variables are defined 
 1. zero _(or minimum?)_ room for interpretation

Reply:
Love the idea of using isolated entry points to solve the cyclic dependencies! TIL. 

I agree with pretty much everything else, too. Just had one minor question regarding this:

```ts
// configs.js
export const ABC = 123
export const XYZ = async () => fetch(...) // async stuff
```

When someone wants to consume `XYZ`, they would have to write `await XYZ()`. Do you think this would be a little confusing since in most cases, people won't expect a variable having a fully uppercase name to be a promise. Maybe we can go with `getXYZ()` for async stuff instead?

Reply:
Perfect. 👌 

```ts
// configs.js
export const ABC = 123;

export const fetchXyz = async () => {
  const XYZ = await fetch(...);
  return { XYZ };
}


// otherfile.ts
import { getXyz } from "./configs"

const doSomething = async () => {
  const { XYZ } = await getXyz();
  console.log({ XYZ });
}
```